flink-ml-lib/src/main/java/org/apache/flink/ml/clustering/agglomerativeclustering/AgglomerativeClustering.java

```diff
@@ -113,18 +110,17 @@ public Table[] transform(Table... inputs) {
                 new OutputTag<Tuple4<Integer, Integer, Double, Integer>>("MERGE_INFO") {};
 
         SingleOutputStreamOperator<Row> output =
-                dataStream.transform(
-                        "doLocalAgglomerativeClustering",
-                        outputTypeInfo,
-                        new LocalAgglomerativeClusteringOperator(
-                                getFeaturesCol(),
-                                getLinkage(),
-                                getDistanceMeasure(),
-                                getNumClusters(),
-                                getDistanceThreshold(),
-                                getComputeFullTree(),
-                                mergeInfoOutputTag));
-        output.getTransformation().setParallelism(1);
+                WindowUtils.windowAll(dataStream, getWindow())
```

Comment:
Does it mean that any algorithm that supports window can only apply the transformation logic with parallelism = 1?

Can we improve its performance by applying local aggregation/transformation on multiple parallelism? Maybe see `AllReduceImpl::allReduceSum` for example.